src/libDirectoryService/DirectoryService.cpp

```diff
@@ -268,6 +269,13 @@ bool DirectoryService::ProcessSetPrimary(const vector<unsigned char>& message,
         m_consensusMyID++;
     }
     m_consensusLeaderID = 0;
+    if (m_mediator.m_currentEpochNum > 1)
```

Comment:
Or, alternatively, since SetPrimary feeds us the Peer of the leader, couldn't we just look through the current `m_DSCommittee` to know the leader index?  Or perhaps we can do both to cross-check?

Reply:
This check is not needed to be honest. Since SetPrimary should only be invoked in epoch 1. I just add this for symmetry.

Reply:
I thought about that too, but wondered if you put that there because it might also be invoked when trying to reboot the whole network (in which case epoch would be X). But yeah I don't think we have startup code for setting `m_currentEpochNum` to whatever is the last block number in persistent storage plus one.

Reply:
I think anyways SetPrimary is contradictory to random ds selection, maybe I can put a warning log instead there if SetPrimary is called not in epoch 1.